src/System.CommandLine/Invocation/CommandHandler.cs

```diff
@@ -282,6 +282,10 @@ public static class CommandHandler
             Func<T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16, Task<int>> action) =>
             HandlerDescriptor.FromDelegate(action).GetCommandHandler();
 
+        public static ICommandHandler FromBindingContext<THandler>()
```

Comment:
It might be helpful for the naming to reflect that the actual handler will be created during binding. One not great idea to open up discussion: `DuringBindingCreateHandlerOfType<T>`

Reply:
`CreateLazyBoundHandler` (to match with the existing `Create`)?